Mathlib/Analysis/SpecialFunctions/Integrals.lean

```diff
@@ -224,10 +225,40 @@ theorem _root_.IntervalIntegrable.log (hf : ContinuousOn f [[a, b]])
     IntervalIntegrable (fun x => log (f x)) μ a b :=
   (ContinuousOn.log hf h).intervalIntegrable
 
+/-- See `intervalIntegrable_log'` for a version without any hypothesis on the interval, but
+assuming the measure is volume. -/
 @[simp]
 theorem intervalIntegrable_log (h : (0 : ℝ) ∉ [[a, b]]) : IntervalIntegrable log μ a b :=
   IntervalIntegrable.log continuousOn_id fun _ hx => ne_of_mem_of_not_mem hx h
 
+/-- The real logarithm is interval integrable (with respect to the volume measure) on every
+interval. See `intervalIntegrable_log` for a version applying to any locally finite measure,
+but with an additional hypothesis on the interval. -/
+@[simp]
+theorem intervalIntegrable_log' :
+  IntervalIntegrable log volume a b := by
+  -- Log is even, so it suffices to consider the case 0 < a and b = 0
+  apply intervalIntegrable_of_even (fun a ↦ Eq.symm (log_neg_eq_log a))
+
+  intro x hx
+  -- Split integral
+  apply IntervalIntegrable.trans (b := 1)
+  · -- Show integrability on [0…1] using non-negativity of the derivative
+    rw [← neg_neg log]
+    apply IntervalIntegrable.neg
+    apply intervalIntegrable_deriv_of_nonneg
+    · exact (continuous_mul_log.continuousOn.sub continuous_id.continuousOn).neg
+    · intro s hs; norm_num at hs
+      convert ((hasDerivAt_mul_log hs.left.ne.symm).sub (hasDerivAt_id s)).neg using 1
+      norm_num
+    · intro s hs; norm_num at hs; simp
+      exact (log_nonpos_iff hs.left).mpr hs.right.le
+  · -- Show integrability on [1…t] by continuity
+    apply ContinuousOn.intervalIntegrable
+    apply (ContinuousOn.mono Real.continuousOn_log)
+    simp
```

Comment:
This is a so-called non-terminal simp: these can make your code harder to maintain. One way to avoid them is replacing them by the equivalent `simp only`. In this case, a better fix could be to combine it with the next line to `simpa using ...`. Can you do that, please?

Reply:
Implemented by @joneugster. Thanks